File: NeetCode/05-Sliding-Window/LongestSubstringWithoutRepeatingCharacters.java
import java.util.*;
class LongestSubstringWithoutRepeatingCharacters {
    public static int lengthOfLongestSubstring(String s) {
        Set<Character> set = new HashSet<>();
        int n = s.length();
        int left = 0;
        int right = 0;
        int maxLength = 0;
        while(right<n){
            if(!set.contains(s.charAt(right))){
                set.add(s.charAt(right));
                maxLength = Math.max(maxLength,right-left+1);
                right++;
            }
            else{
                set.remove(s.charAt(left));
                left++;
            }
        }
        return maxLength;
    }
    public static void main(String[] args){
        Scanner scan = new Scanner(System.in);
        String s = scan.nextLine();
        System.out.println(lengthOfLongestSubstring(s));
        scan.close();
    }
}

File: NeetCode/05-Sliding-Window/LongestSubstringWithoutRepeatingCharacters.cpp
#include <iostream>
#include <string>
#include <unordered_set>
#include <algorithm>

using namespace std;
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        unordered_set<char> mySet;
        int n = s.size();
        int left = 0;
        int right = 0;
        int maxLength = 0;
        while(right<n){
            if(mySet.find(s[right]) == mySet.end()){
                mySet.insert(s[right]);
                maxLength = max(maxLength,right-left+1);
                right++;
            }
            else{
                mySet.erase(s[left]);
                left++;
            }
        }
        return maxLength;
    }
};
int main(){
    string s;
    getline(cin,s);
    Solution sol;
    cout << sol.lengthOfLongestSubstring(s);
}